app.py

```diff
@@ -329,17 +331,23 @@ def list_data_objects(**kwargs):
     :rtype: ListDataObjectsResponse
     """
     req_body = app.current_request.query_params or {}
-    page_token = req_body.get('page_token', 0)
     per_page = int(req_body.get('page_size', 10))
-    if req_body.get('alias', None):
-        results = azul_match_alias(index=INDEXES['data_obj'],
-                                   alias=req_body['alias'], size=per_page + 1,
-                                   from_=page_token if page_token != 0 else None)
-    else:
-        results = es_query(query={}, index=INDEXES['data_obj'], size=per_page + 1)
+
+    # Build the query. If multiple criteria are specified, returned objects
+    # should match all of the provided criteria (logical AND).
+    query = {'query': {}}
+    if 'page_token' in req_body:
+        query['from'] = req_body['page_token'] or 0
+    if 'alias' in req_body:
+        query['query']['term'] = {}
+        query['query']['term']['aliases.keyword'] = req_body['alias']
+    else:  # if no query parameters are provided
+        query['query']['match_all'] = {}
+    results = es_query(query={}, body=query, index=INDEXES['data_obj'], size=per_page + 1)
```

Comment:
Is this the only place es_query is used? should query not be a default argument?

Reply:
No, `es_query` is used throughout the lambda, and `query` is generally expected with any `es_query` call

Reply:
right, it seems like you would necessarily need a query argument. So then I was wondering why it's an optional argument with a default value. Maybe this is out of scope and unnecessary, but I'm just asking the question

Reply:
That's a good point, and I think it looks bad because I designed it poorly. I've refactored es_query to be better